### ISSUE TITLE
Switching between red and Green alert daily for no obvious reason

### ISSUE THREAD

**Issue by jwildeboer** [corona-warn-app/cwa-app-ios#4351]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for iOS only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

## Technical details

- Device name: iPhone 11
- iOS version: 15.4
- App version: 2.18, 2.19

## Describe the bug

Since at least four days now, risk calculation for the 5th of march switches between "low risk" and "increased risk" with almost every new risk calculation. This makes the status switch from Green to red, causing uncertainty about what the risk really is. 

It 

## Steps to reproduce the issue

In the morning:

![image](https://user-images.githubusercontent.com/873028/158991714-bbf902ad-5707-463c-b546-b36284cba825.jpeg)

![image](https://user-images.githubusercontent.com/873028/158991818-4b2ce8d0-7fda-45a4-9041-530b09d0e54b.jpeg)

A few hours later:

![image](https://user-images.githubusercontent.com/873028/158991004-963bbb23-69c8-4225-896a-a5ed8581c1b5.png)

![image](https://user-images.githubusercontent.com/873028/158991033-201f7afb-878f-4ab5-a110-41e18400c379.jpeg)

I have created and uploaded an error report that contains two of these events. 

![image](https://user-images.githubusercontent.com/873028/158991281-0d603ff4-56e3-4a64-8877-75df311e4c40.jpeg)

## Additional context

---
Internal Tracking ID: [EXPOSUREAPP-12367](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12367)

**Comment by Ein-Tim** [comment]
@mlenkeit 

This reminds me a bit of https://github.com/corona-warn-app/cwa-app-android/issues/4925.

Could you check the log to investigate this, please?

**Comment by Ein-Tim** [comment]
FYI: Twitter user experiencing something very similar: https://twitter.com/gerdkurschner/status/1504905213588742144?s=21

**Comment by jwildeboer** [comment]
FTR - it again switched two times in the last 24 hours. Right now it's green again. And it's always the 5th of march that keeps changing. I suspect some weird rounding error during risk calculation. For me at least it will be over tomorrow when the 5th of march falls out of the 14 day period :) 

![image](https://user-images.githubusercontent.com/873028/159117001-f41f12df-1e14-4765-80cc-09b0fde30ee4.jpeg)

![image](https://user-images.githubusercontent.com/873028/159117006-440e64e7-be34-41ef-a670-56d1445fa02a.jpeg)

**Comment by Ein-Tim** [comment]
@thomasaugsten Are you looking into this issue already?

**Comment by thomasaugsten** [comment]
A error report would be helpful to identify the root cause

**Comment by Ein-Tim** [comment]
@thomasaugsten 

The ID is included in @jwildeboer's OP.

**Comment by jwildeboer** [comment]
@thomasaugsten an Error report that should have captured two of these switching events was created and shared right before I opened the issue. ![image](https://user-images.githubusercontent.com/873028/159152630-33c367f9-a345-4f97-b875-08bb1a0512c7.jpeg)

**Comment by Ein-Tim** [comment]
Was anything helpful found in the log?

**Comment by jwildeboer** [comment]
After 5 days still no real progress :( A bit disappointing. I thought I did everything to help the developers, but I guess I somehow didn't do enough. As the offending day is long removed from my risk calculation I cannot be of help anymore anyway. I hope someone picks this up and fixes it, as doesn't only happen to me.

**Comment by thomasaugsten** [comment]
We cannot see explicit in the error log. We see a lot of failing risk check in the background.
We hope with iOS 2.19.1 with delta submission to the ENF like on Android we can optimize the risk check and maybe this behavior will disappear but in parallel we will try to find out why a failed risk check leads to a green card.

**Comment by mlenkeit** [comment]
@jwildeboer don't be too hard on yourself 😉  Analysis of the error log is still going on. We have found an unusual sequence log entries related to the risk check but are still trying to understand the root cause. 

For all we know so far, it does not seem to be related to https://github.com/corona-warn-app/cwa-app-android/issues/4925.

**Comment by Ein-Tim** [comment]
@mlenkeit 

Any update you can share with us?

**Comment by mlenkeit** [comment]
> @mlenkeit
> 
> Any update you can share with us?

@Ein-Tim the unusual sequence in the error log seems to be related to the code execution in the background job and premature termination of the job. The problem seems to appear in the background job following such a sequence.

This is very difficult to reproduce and from the logs itself, we are still struggling to understand the actual root cause.

Additional error reports might help us to track down the problem.

**Comment by jwildeboer** [comment]
Will start another error report and upload it. But ATM I don't have the switching between red and Green problem.

**Comment by thomasaugsten** [comment]
How often is the alert switched in this error report?

**Comment by Ein-Tim** [comment]
@thomasaugsten See https://github.com/corona-warn-app/cwa-app-ios/issues/4351#issuecomment-1082091991

**Comment by larswmh** [comment]
> Since at least four days now

@jwildeboer Can you remember the dates for those four days? In log D0FDC3C128C1B0E80412 (18th of March) we were only able to see entries for 2022-03-17 and 2022-03-18. Did the "switching" between risk levels happened on these days?

Your error reports from 30th and 31st of March have been forwarded to the internal ticket.

**Comment by Ein-Tim** [comment]
Twitter user reporting something very similar: https://twitter.com/ChristianeKas13/status/1513785078769463298?s=20&t=B4KFrF2Xl2SVPLzEOkA8Dg

**Comment by dsarkar** [comment]
@Ein-Tim Thanks, forwarded to internal ticket.

**Comment by Ein-Tim** [comment]
Hey @dsarkar, could you please check if the PR https://github.com/corona-warn-app/cwa-app-ios/pull/4535 by @ArturFriesen could also fix this issue?

**Comment by larswmh** [comment]
Closing this issue as it was fixed in the abovementioned PR.
